Rollbar/Telemetry/TelemetryCollector.cs

```diff
@@ -0,0 +1,275 @@
+                    CompleteProcessing();
+                }
+                return this;
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether this instance is auto-collecting.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if this instance is auto-collecting; otherwise, <c>false</c>.
+        /// </value>
+        public bool IsAutocollecting
+        {
+            get
+            {
+                lock(this._syncRoot)
+                {
+                    return !(this._cancellationTokenSource == null || this._cancellationTokenSource.IsCancellationRequested);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the queue depth.
+        /// </summary>
+        /// <value>
+        /// The queue depth.
+        /// </value>
+        public int QueueDepth { get { return this._telemetryQueue.QueueDepth; } }
+
+        private readonly TelemetryConfig _config;
+        private readonly TelemetryQueue _telemetryQueue = new TelemetryQueue();
+        private Thread _telemetryThread = null;
+        private CancellationTokenSource _cancellationTokenSource = null;
+        private readonly object _syncRoot = new object();
+
+        private void _config_Reconfigured(object sender, EventArgs e)
+        {
+            this._telemetryQueue.QueueDepth = this._config.TelemetryQueueDepth;
+
+            if (this.IsAutocollecting && this._config.TelemetryAutoCollectionInterval == TimeSpan.Zero)
+            {
+                this.StopAutocollection(false);
+            }
+        }
+
+        private void CompleteProcessing()
+        {
+            if (this._cancellationTokenSource == null)
+            {
+                return;
+            }
+
+            Debug.WriteLine("Entering " + this.GetType().FullName + "." + nameof(this.CompleteProcessing) + "() method...");
+            this._cancellationTokenSource.Dispose();
+            this._cancellationTokenSource = null;
+            this._telemetryThread = null;
+        }
+
+        private void AutoCollectTelemetry()
+        {
+            //TBD...
+        }
+
+        private void KeepCollectingTelemetry(object data)
+        {
+            //for now, until AutoCollectTelemetry() is implemented:
+            return;
+
+            CancellationToken cancellationToken = (CancellationToken)data;
+
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                try
+                {
+                    this.AutoCollectTelemetry();
+                }
+#pragma warning disable CS0168 // Variable is declared but never used
+                catch (System.Threading.ThreadAbortException tae)
+                {
+                    return;
+                }
+                catch (System.Exception ex)
+                {
+                    string msg = ex.Message;
+                    //TODO: do we want to direct the exception 
```

Comment:
We don't want to report anything to us, this should just be a no-op, or at worst a debug log message to the console.

Reply:
ok, adding tracing of the exception...